public/controllers/agent/register-agent/steps/server-address.tsx

```diff
@@ -0,0 +1,109 @@
+import { EuiComboBox, EuiComboBoxOptionOption, EuiHighlight, EuiText } from '@elastic/eui';
+import React, { useEffect, useState } from 'react';
+import { getMasterNode, parseNodeIPs } from '../utils';
+
+type Props = {
+  onChange: (value: string) => void;
+  fetchOptions: () => Promise<EuiComboBoxOptionOption<string>[]>;
+};
+
+export type ServerAddressOptions = EuiComboBoxOptionOption<any> & { nodetype?: string };
+
+export default function ServerAddress(props: Props) {
+  const { onChange, fetchOptions } = props;
+  const [nodeIPs, setNodeIPs] = useState<ServerAddressOptions[]>([]);
+  const [selectedNodeIPs, setSelectedNodeIPs] = useState<ServerAddressOptions[]>([]);
+  const [isLoading, setIsLoading] = useState(false);
+
+  useEffect(() => {
+    initialize();
+  }, []);
+
+  /**
+   * Fetches the node IPs (options) and sets the state
+   */
+  const initialize = async () => {
+    if(!fetchOptions) throw new Error('fetchOptions is required');
+    
+    try {
+      setIsLoading(true);
+      const nodeIps = await fetchOptions();
+      setNodeIPs(nodeIps);
+      const defaultNode = getMasterNode(nodeIps);
+      if (defaultNode.length > 0) handleOnChange(defaultNode);
```

Comment:
According to our [style guide](https://github.com/wazuh/wazuh-kibana-app/blob/master/STYLEGUIDE.md#always-use-braces-for-conditionals-and-loops), we should always wear brackets.